### DISCUSSION TITLE
Web UI or Console on Unraid

### DISCUSSION THREAD

**Discussion by BorgusBorg** [itzg/docker-minecraft-server#1327] [Q&A]
New to unraid and docker, set up an unraid server and have had Minecraft running nicely for a while now. Only thing I can't figure out is how to access the Server/Admin console so I can put in commands (for example).  I can see live logs by using the logs item in the context menu in unraid, but I can't interact, and I can open the docker console, but that doesn't let me connect to rcon-cli. What i'd like is to be able to use the "web UI" option from the context menu like my other dockers to access the server/admin console, to let me see the logs live as well as to pass commands to the server without being in-game.

Let me know what, if any, info you need from me to help out please.

Thanks!

**Comment by itzg** [comment]
I'm not familiar with the Unraid interface, so I was wondering if you could provide a screenshot of what it gives you when choosing the docker console?

**Reply by BorgusBorg** [reply]
Hi, I attached what I think you are looking for? 
![console](https://user-images.githubusercontent.com/98989938/152470188-17e2be7b-f9af-4b13-9e97-1dc90e6df57e.PNG)

When I try and connect to the rcon in that console, I get this in the logs (I had tried many times in a row here, not just the one time like above)
![logs](https://user-images.githubusercontent.com/98989938/152470868-e3317498-2dd1-499f-ad25-8cf460eaca50.PNG)

as well as my docker setup
![DockerSetup2](https://user-images.githubusercontent.com/98989938/152470185-dec31279-8679-40a8-a1e4-28b76a92bb83.PNG)
![DockerSetup1](https://user-images.githubusercontent.com/98989938/152470187-d0df875c-8925-425b-9752-d43f4efae239.PNG)

If I try and access the webUI, I get the same error in the logs and it never loads.

**Reply by itzg** [reply]
Thanks, yes that helped. That is an usual behavior for the console, but then it seems like each of the NAS vendors each re-invent parts of Docker. 

It looks like their console is not propagating the environment variables, which is actually a reasonable security measure. You can pass the rcon password on the command as 

```
rcon-cli --password <password>
```

**Reply by BorgusBorg** [reply]
Hi itzg,

Thank you for taking the time to respond and look into this. I just tried this with no luck unfortunately. 
![Console Log](https://user-images.githubusercontent.com/98989938/152572248-7144a005-2aab-44e6-9f78-1531042dc93d.PNG)

Would you have any other thoughts by any chance?

Thanks!

**Reply by BorgusBorg** [reply]
I figured it out - I had set a different password in server.properties than I had in the docker setup, because the docker doesn't overwrite the server properties, the password I was using was the wrong one! This thread of yours helped me out! https://gitter.im/itzg/dockerfiles?at=5c55c5c878e1ed4103ce811b 

My next step is trying to figure out how to get the interactive/color console, would you have any thought on that?

**Reply by itzg** [reply]
> My next step is trying to figure out how to get the interactive/color console, would you have any thought on that?

That will mainly depend on Unraid's support of the equivalent of `docker attach`.
